lib/opal/cli_runners/applescript.rb

```diff
@@ -7,52 +7,28 @@ def initialize(options)
         unless system('which osalang > /dev/null')
           raise MissingJavaScriptSupport, 'JavaScript Automation is only supported by OS X Yosemite and above.'
         end
-
-        @output = options.fetch(:output, $stdout)
+        command_options = {:name => 'applescript', :cmd => ['osascript', '-l', 'JavaScript']}
+        command_options[:options] = options
+        @cmd = Cmd.new(command_options)
       end
-      attr_reader :output, :exit_status
 
       def puts(*args)
```

Comment:
I wasn't able to determine if this method is used or not...

Reply:
`exit_status` is surely used here: https://github.com/opal/opal/blob/319a38463530a0da527fb190a3fb361608bc303e/lib/opal/cli_runners.rb#L59-L67

But of course is rewritten with the new API it's not needed

Reply:
Speaking of `exit_status`, on the master branch the return code is `0` even when a spec fails... do you reproduce this issue ?

1. edit `spec/opal/core/array/intersection_spec.rb` to create a failure
1. run `bundle exec rake mspec_opal_nodejs PATTERN=spec/opal/core/array/intersection_spec.rb`
1. `echo $?` is `0`